app/scripts/views/settings.js

```diff
@@ -3,114 +3,296 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 define([
+  'jquery',
+  'modal',
   'cocktail',
+  'lib/promise',
   'lib/session',
-  'views/form',
   'views/base',
   'views/mixins/avatar-mixin',
+  'views/settings/avatar',
+  'views/settings/avatar_change',
+  'views/settings/avatar_crop',
+  'views/settings/avatar_camera',
+  'views/settings/avatar_gravatar',
+  'views/settings/gravatar_permissions',
+  'views/settings/communication_preferences',
+  'views/settings/change_password',
+  'views/settings/delete_account',
+  'views/settings/display_name',
   'views/mixins/settings-mixin',
+  'views/decorators/allow_only_one_submit',
   'stache!templates/settings'
 ],
-function (Cocktail, Session, FormView, BaseView, AvatarMixin,
-  SettingsMixin, Template) {
+function ($, modal, Cocktail, p, Session, BaseView, AvatarMixin,
+  AvatarView, AvatarChangeView, AvatarCropView, AvatarCameraView, GravatarView,
+  GravatarPermissionsView, CommunicationPreferencesView, ChangePasswordView,
+  DeleteAccountView, DisplayNameView, SettingsMixin, allowOnlyOneSubmit,
+  Template) {
   'use strict';
 
   var t = BaseView.t;
 
-  var View = FormView.extend({
+  var PANEL_VIEWS = [
+    AvatarView,
+    DisplayNameView,
+    CommunicationPreferencesView,
+    ChangePasswordView,
+    DeleteAccountView,
+    AvatarChangeView,
+    AvatarCropView,
+    AvatarCameraView,
+    GravatarView,
+    GravatarPermissionsView
+  ];
+
+  function renderView(view) {
+    return view.render()
+      .then(function (shown) {
+        if (! shown) {
+          view.destroy(true);
+          return;
+        }
+        view.afterVisible();
+
+        return view;
+      });
+  }
+
+  var View = BaseView.extend({
     template: Template,
     className: 'settings',
 
+    FADE_OUT_SETTINGS_MS: 250,
+
     initialize: function (options) {
       options = options || {};
 
       this._able = options.able;
+      this._subViewToShow = options.subView;
+      this._panelViews = options.panelViews || PANEL_VIEWS;
+
+      this.router.on(this.router.NAVIGATE_FROM_SUBVIEW, this._onNavigateFromSubview.bind(this));
     },
 
     context: function () {
       var account = this.getSignedInAccount();
-      var email = account.get('email');
 
       return {
-        email: email,
+        username: account.get('email'),
```

Comment:
At one point we talked about showing the user's displayName if that is set, otherwise the email address.